uxarray/grid.py

```diff
@@ -380,6 +384,65 @@ def integrate(self, var_ds, quadrature_rule="triangular", order=4):
 
         return integral
 
+    def _build_edge_node_connectivity(self):
+        """Constructs the UGRID connectivity variable (``Mesh2_edge_nodes``)
+        and stores it within the internal (``Grid.ds``) and through the
+        attribute (``Grid.Mesh2_edge_nodes``).
+
+        Additionally, the attributes (``inverse_indices``) and
+        (``fill_value_mask``) are stored for constructing other
+        connectivity variables.
+        """
+
+        # padded face nodes: [nMesh2_face x nMaxMesh2_face_nodes + 1]
+        padded_face_nodes = _close_face_nodes(self.Mesh2_face_nodes.values,
+                                              self.nMesh2_face,
+                                              self.nMaxMesh2_face_nodes)
+
+        # construct an array of empty edge nodes where each entry is a pair of indices
+        edge_nodes = np.empty((self.nMesh2_face * self.nMaxMesh2_face_nodes, 2),
+                              dtype=INT_DTYPE)
+
+        # first index includes starting node up to non-padded value
+        edge_nodes[:, 0] = padded_face_nodes[:, :-1].ravel()
+
+        # second index includes second node up to padded value
+        edge_nodes[:, 1] = padded_face_nodes[:, 1:].ravel()
+
+        # all edge nodes that contain a fill value
+        fill_value_mask = np.logical_or(edge_nodes[:, 0] == INT_FILL_VALUE,
+                                        edge_nodes[:, 1] == INT_FILL_VALUE)
+
+        # all edge nodes that do not contain a fill value
+        non_fill_value_mask = np.logical_not(fill_value_mask)
+
+        # filter out all invalid edges
+        edge_nodes = edge_nodes[non_fill_value_mask]
+
+        # sorted edge nodes
+        edge_nodes.sort(axis=1)
+
+        # unique edge nodes
+        edge_nodes_unique, inverse_indices = np.unique(edge_nodes,
+                                                       return_inverse=True,
+                                                       axis=0)
+        # add mesh2_edge_nodes to internal dataset
+        self.ds['Mesh2_edge_nodes'] = xr.DataArray(
+            edge_nodes_unique,
+            dims=["nMesh2_edge", "Two"],
+            attrs={
+                "cf_role":
+                    "edge_node_connectivity",
+                "long_name":
+                    "Maps every edge to the two nodes that it connects",
+                "start_index":
+                    INT_DTYPE(0),
+                "inverse_indices":
+                    inverse_indices,
+                "fill_value_mask":
+                    fill_value_mask
+            })
```

Comment:
The addition of this DataArray in our `Grid.ds` is breaking the `TestIntegrate.test_calculate_total_face_area_triangle` cases. It causes the the `face_area` to be improperly calculated.

@rajeeja Do you have any idea why this is breaking? I attempted to do some digging, but wasn't able to find anything.  All other test cases pass locally.

Reply:
strange, how different are the values?

Reply:
The same occurs when calculating `area_triangular`

Reply:
Has your [commit](https://github.com/UXARRAY/uxarray/pull/263/commits/96d66d4c65604880b437d4a370b969ad502ec0ac) fixed the issue? no failures reported now.

Reply:
It appears to be working now with the latest merges